src/MatrixTwitter.js

```diff
@@ -0,0 +1,1040 @@
+ * @param  config. Bridge configuration (currently only make use of the auth
+*  information)
+ * @param  config.app_auth.consumer_key Twitter consumer key
+ * @param  config.app_auth.consumer_secret Twitter consumer secret
+ * @param  {TwitterDB}       storage
+ */
+var MatrixTwitter = function (bridge, config, storage) {
+  this.app_auth = config.app_auth;
+  this.app_twitter = null;
+  this.tclients = new Map(); // {'@userid':TwitterClient}
+  this.storage = storage;
+
+  this.timeline_queue = [];
+  this.timeline_intervalID = null;
+
+  this.hashtag_intervalID = null;
+  this.hashtag_queue = [];
+
+  this.user_streams = new Map(); // {'@userid':TwitterStream}
+
+  this.processed_tweets = new ProcessedTweetList(256, 32);  //This will contain all the tweet IDs of things we don't want to repeat.
+  this.sent_dms = new ProcessedTweetList(1, 1);  //This will contain the body of the DM posted to a room to avoid reposting it.
+  this.msg_queue = [];
+  this.msg_queue_intervalID = null;
+  this._bridge = bridge;
+};
+
+
+/**
+ * MatrixTwitter.prototype.start - Starts the timers for polling the API and
+ * authenticates the application with Twitter.
+ * @function
+ * @return {Promise}  A promise that returns when authentication succeeds.
+ */
+MatrixTwitter.prototype.start = function () {
+  if(this.app_twitter != null) {
+    log.warn("Twitter",  "Attempted to call start() while having been started previously.");
+    return
+  }
+
+  return this._get_bearer_token().then((token) => {
+    this.app_auth.bearer_token = token;
+    log.info('Twitter', 'Retrieved token');
+    this.app_twitter = new Twitter(this.app_auth);
+    this.msg_queue_intervalID = setInterval(() => {
+      this._process_head_of_msg_queue();
+    }, TWITTER_MSG_QUEUE_INTERVAL_MS);
+
+    this.start_timeline();
+    this.start_hashtag();
+
+  }).catch((error) => {
+    log.error('Twitter', 'Error trying to retrieve bearer token:', error);
+    throw "Couldn't get a bearer token for Twitter AS.";
+  });
+}
+
+MatrixTwitter.prototype._get_bearer_http = function () {
+  return new Promise( (resolve, reject) => {
+    var key = this.app_auth.consumer_key + ":" + this.app_auth.consumer_secret;
+    key = Buffer.from(key, 'ascii').toString('base64');
+    var options = {
+      url: "https://api.twitter.com/oauth2/token",
+      headers: {
+        'Authorization': "Basic " + key
+      },
+      form: "grant_type=client_credentials",
+      contentType: "application/x-www-form-urlencoded;charset=UTF-8"
+    };
+    Request.post(options, function (error, response, body) {
+      if (error) {
+        reject(error);
+      } else if (response.statusCode !== 200) {
+        reject("Response to bearer token request returned non OK")
+        log.error("Twitter",
+              "Body of response:%s\nStatuscode of respnse:%s",
+              body,
+              response.statusCode
+            );
+      } else {
+        try {
+          var jsonresponse = JSON.parse(body);
```

Comment:
The try/catch only needs to be around this line, so you can outdent the rest of this else block and save some space.